h2o-algos/src/main/java/hex/gam/GAMModel.java

```diff
@@ -0,0 +1,445 @@
+package hex.gam;
+
+import hex.*;
+import hex.deeplearning.DeepLearningModel;
+import hex.gam.MatrixFrameUtils.AddGamColumns;
+import hex.glm.GLM;
+import hex.glm.GLMModel;
+import hex.glm.GLMModel.GLMParameters.Family;
+import hex.glm.GLMModel.GLMParameters.GLMType;
+import hex.glm.GLMModel.GLMParameters.Link;
+import hex.glm.GLMModel.GLMParameters.Solver;
+import water.*;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+import water.udf.CFuncRef;
+import water.util.ArrayUtils;
+import water.util.Log;
+
+import java.io.Serializable;
+import java.util.Arrays;
+
+import static hex.gam.MatrixFrameUtils.GamUtils.equalColNames;
+import static hex.glm.GLMModel.GLMParameters.MissingValuesHandling;
+
+public class GAMModel extends Model<GAMModel, GAMModel.GAMParameters, GAMModel.GAMModelOutput> {
+  public boolean _centerGAM = false;  // true if centering is needed for training dataset
+  public String[][] _gamColNames; // store column names only for GAM columns
+  public String[][] _gamColNamesCenter; // store column names only for GAM columns after decentering
+  public Key<Frame>[] _gamFrameKeys;
+  public Key<Frame>[] _gamFrameKeysCenter;
+  public int _nclass; // 2 for binomial, > 2 for multinomial and ordinal
+
+  @Override public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+    return null;
```

Comment:
probably wnat to fix that?

Reply:
Yeah, I added the modelMetrics stuff in.